src/models/environment.ts

```diff
@@ -243,14 +254,43 @@ function createEnvironmentTooltip(resource: Environment): MarkdownString {
         `\n\n[$(${IconNames.CONFLUENT_LOGO}) Open in Confluent Cloud](${ccloudEnv.ccloudUrl})`,
       );
   } else if (isDirectResource) {
+    // check for any resources that the sidecar reported a `FAILED` connection status.
+    // ideally, the ResourceViewProvider would react to events pushed by the ConnectionStateWatcher
+    // and update the environments' `kafkaConnectionFailed` and `schemaRegistryConnectionFailed`
+    // properties, but in the event we didn't get those websocket events (e.g. new workspace),
+    // we can check to see if they're just "missing" based on the expected configuration(s)
+    const directEnv = resource as DirectEnvironment;
     const { missingKafka, missingSR } = checkForMissingResources(resource);
-    if (missingKafka || missingSR) {
-      const missing = [];
-      if (missingKafka) missing.push("Kafka");
-      if (missingSR) missing.push("Schema Registry");
+
+    const failedResources = [];
+    const missingResources = [];
+
+    if (directEnv.kafkaConnectionFailed) {
+      failedResources.push(`**Kafka**: ${directEnv.kafkaConnectionFailed}`);
+    } else if (missingKafka) {
+      missingResources.push("Kafka");
+    }
+
+    if (directEnv.schemaRegistryConnectionFailed) {
+      failedResources.push(`**Schema Registry**: ${directEnv.schemaRegistryConnectionFailed}`);
+    } else if (missingSR) {
+      missingResources.push("Schema Registry");
+    }
+
+    if (failedResources.length) {
+      tooltip.appendMarkdown("\n\n---").appendMarkdown("\n\n$(error) **Unable to connect to**:");
+      failedResources.forEach((error) => {
+        tooltip.appendMarkdown(`\n\n- ${error}`);
+      });
+      // provide a command URI as a markdown link
+      const commandUri = Uri.parse(
+        `command:confluent.connections.direct.edit?${encodeURIComponent(JSON.stringify([resource.connectionId]))}`,
+      );
+      tooltip.appendMarkdown(`\n\n[View Connection Details](${commandUri})`);
+    } else if (missingResources.length) {
       tooltip
         .appendMarkdown("\n\n---")
-        .appendMarkdown(`\n\n$(error) Unable to connect to ${missing.join(" and ")}.`);
+        .appendMarkdown(`\n\n$(error) Unable to connect to ${missingResources.join(" and ")}.`);
```

Comment:
This shouldn't happen anymore, but is the fallback in the event we can't cache connection status(es) and/or miss websocket event(s) for the associated connection state (but still know that the spec doesn't match what returned from the GraphQL query).